src/components/app/data/services/academies.js

```diff
@@ -11,14 +13,12 @@ export async function fetchAcademies(enterpriseUUID, options = {}) {
   const url = `${getConfig().ENTERPRISE_CATALOG_API_BASE_URL}/api/v1/academies?${queryParams.toString()}`;
 
   try {
-    const response = await getAuthenticatedHttpClient().get(url);
-    const data = camelCaseObject(response.data);
-    return data.results ?? [];
+    const { results } = await fetchPaginatedData(url);
+    return results;
   } catch (error) {
-    const errorResponseStatusCode = getErrorResponseStatusCode(error);
-    if (errorResponseStatusCode === 404) {
-      return [];
+    if (getErrorResponseStatusCode(error) !== 404) {
+      logError(error);
     }
-    throw error;
+    return [];
   }
```

Comment:
[inform] Same general treatment for all API calls made during route loaders.

That is, if the error is expected (e.g., known 404 as in this case), return valid data albeit empty. If the error is NOT expected (e.g., 500), still return valid/empty data but ensure the error is logged for observability,.